browse/services/object_store/fileobj.py

```diff
@@ -187,10 +209,10 @@ def name(self) -> str:
     def exists(self) -> bool:
         return self._fileobj.exists()
 
-    @contextmanager
-    def open(self, *args, **kwargs) -> IO:  # type: ignore
-        with self._fileobj.open(mode="rb") as zipped:
-            yield gzip.GzipFile(filename="", mode="rb", fileobj=zipped)
+    def open(self, mode:str) -> BinaryMinimalFile:
+        return gzip.GzipFile(filename="",
+                             mode=mode,
+                             fileobj=self._fileobj.open(mode))
```

Comment:
Again, I'm not sure what return type you want here, but BinaryMinimalFile seems wrong.

Reply:
BinaryMinimalFile is a protocol that I'm using to inform Mypy of the minimal interface I need to get range requests to work.
At this line mypy should be saying "Okay it's returning BinaryMinimalFile, does the object returned meet that interface?"